### PR TITLE
data models for race and ethnicity

### PR DESCRIPTION
**Pull request by saikumar9**
# PR Checklist

Please check if your PR fulfills the following requirements
- [x] The title follows our [guidelines](https://github.com/ideacrew/enroll/blob/trunk/CONTRIBUTING.md#commit
)
- [x] Tests for the changes have been added (for bugfixes/features)

# PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature (requires Feature flag)
- [ ] Data fix or migration (inert code, no impact until run)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Dependency updates (e.g., add a new gem or updated version)

# What is the ticket # detailing the issue?

Ticket: [Teal Team 6 Development - 187040832](https://www.pivotaltracker.com/story/show/187040832)

# A brief description of the changes

Current behavior: N/A

New behavior: New Data Models added for persisting Race and Ethnicity information. This is related to CRs `232 (IVL) + 246 (SHOP) Improve collection of race and ethnicity`.

# Feature Flag

For all new feature development, a feature flag is required to control the exposure of the feature to our end users. A feature flag needs a corresponding environment variable to initialize the state of the flag. Please share the name of the environment variable below that would enable/disable the feature and which client(s) it applies to.

Variable name:

- [ ] DC
- [ ] ME

# Additional Context
Include any additional context that may be relevant to the peer review process.

# AppScan CodeSweep Failure
In the event of a failed check on the AppScan CodeSweep step of our GitHub Actions workflow, please review the False Positive protocol outlined here: appscan_codesweep/CODESWEEP_FALSE_POSITIVES_README.MD

Add all required notes to this section if the failure is a suspected false positive.

### CODE REVIEW

**Review by ymhari — APPROVED** [review]

File: Gemfile.lock [hunk 1]
```diff
@@ -548,10 +548,10 @@ GEM
     fast_jsonapi (1.5)
       activesupport (>= 4.2)
     ffaker (2.18.0)
+    ffi (1.15.5)
```
Comment (mdkaraman) [inline Gemfile.lock:551]: Is this change intentional?

File: app/models/person_demographics_group.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  include Mongoid::Timestamps
+
+  # @!attribute [rw] demographable
+  #   @return [Object] The object that this PersonDemographicsGroup instance is associated with.
+  #   This is a polymorphic association, so it can be any object that declares itself as 'demographable'.
+  embedded_in :demographable, polymorphic: true
+
+  # @!attribute [rw] ethnicities
+  #   @return [Array<Ethnicity>] The ethnicities associated with this PersonDemographicsGroup.
+  embeds_many :ethnicities, class_name: 'Ethnicity', cascade_callbacks: true, validate: true
```
Comment (mdkaraman) [inline app/models/person_demographics_group.rb:16]: Should this be commented out for now?  Or at least have a comment that it is not in use yet?

File: app/models/person_demographics_group.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  # @!attribute [rw] races
+  #   @return [Array<Race>] The races associated with this PersonDemographicsGroup.
+  embeds_many :races, class_name: 'Race', cascade_callbacks: true, validate: true
```
Comment (mdkaraman) [inline app/models/person_demographics_group.rb:20]: Same comment as above for ethnicities field.  Without context someone may be confused if they should pull/persist R/E data using these models.

File: app/models/race.rb [hunk 4]
```diff
@@ -0,0 +1,148 @@
+# frozen_string_literal: true
+
```
Comment (mdkaraman) [inline app/models/race.rb:2]: Consider a top level comment that this is part of an intended feature and not the current source of truth for race data.

File: app/models/ethnicity.rb [hunk 5]
```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
```
Comment (mdkaraman) [inline app/models/ethnicity.rb:2]: Consider a top level comment that this is part of an intended feature and not the current source of truth for ethnicity data.

**Review by mdkaraman — COMMENTED** [review]
@saikumar9 let me know if this feedback makes sense to you.  My only concern is that we are adding the R/E models as if they will be supported at the same time as the AliveStatus model.  This could be confusing without some comments.
